### PR TITLE
Update for main package version 0.14 and PHP7

### PR DESCRIPTION
**Pull request by mallardduck**
Per title, this PR fixes this git version by updating for pending changes in https://github.com/tomzx/gitter/pull/4.

Once the above PR is merged into the gitter this package uses, then this PR can be updated and then merged in. Mainly to ensure the compser.lock is updated fully.

### CODE REVIEW

**Comment by mallardduck** [comment]
This addresses the issue opened here: https://github.com/tomzx/php-semver-checker-git/issues/29

File: composer.json [hunk 1]
```diff
@@ -27,16 +27,16 @@
...
 		"symfony/console": "^2.7|^3.0|^4.0",
 		"tomzx/finder": "^0.1",
-		"tomzx/php-semver-checker": "^0.12",
+		"tomzx/php-semver-checker": "^0.14",
 		"vierbergenlars/php-semver": "^3.0"
 	},
 	"require-dev": {
 		"mockery/mockery": "^1.0",
-		"phpunit/phpunit": "^4.0|^5.0"
+		"phpunit/phpunit": "^7.0|^8.0|^9.0"
```
Comment (tomzx) [inline composer.json:39]: **SHOULD**
PHPUnit 8 and 9 are currently supported, we should drop ^7.0.

File: src/PHPSemVerCheckerGit/Console/Command/SuggestCommand.php [hunk 2]
```diff
@@ -21,222 +21,222 @@
...
-				if (SemanticVersion::eq($versionTag, $tag)) {
-					return $tag;
-				}
-			} catch (RuntimeException $e) {
-				// Do nothing
-			}
-		}
-		return null;
-	}
+    /**
```
Comment (tomzx) [inline src/PHPSemVerCheckerGit/Console/Command/SuggestCommand.php:24]: **MUST**
Let's keep the tab indentation. Migration to PSR-2 will be done in a separate PR.

**Review by tomzx — COMMENTED** [review]
Overall PR looks good to me. I'm in the process of updating the dependencies for PHP >=7.2 so I might cherry-pick part of your changes instead of the whole PR.

File: composer.json (hunk 1 above)
Comment (mallardduck) [reply composer.json:39]: Good deal - I think that this may have been to match tomzx/gitter changes.
So I can rework that PR for PHP >=7.2 as well and drop PHPunit 7.0 from both.

File: composer.json [hunk 3]
```diff
@@ -23,20 +23,20 @@
 	"repositories": [
 		{
 			"type": "vcs",
-			"url": "https://github.com/tomzx/gitter"
+			"url": "https://github.com/mallardduck/gitter"
```
Comment (mallardduck) [inline composer.json:26]: Note: This would be reverted once this PR https://github.com/tomzx/gitter/pull/4/files is merged in.

File: composer.json [hunk 4]
```diff
@@ -23,20 +23,20 @@
...
 		}
 	],
 	"require": {
-		"php": ">=5.6.0",
-		"klaussilveira/gitter": "dev-php-semver-checker",
+		"php": ">=7.2.29",
+		"klaussilveira/gitter": "dev-7.4-update",
```
Comment (mallardduck) [inline composer.json:31]: Note: This would be reverted once this PR https://github.com/tomzx/gitter/pull/4/files is merged in.

File: src/PHPSemVerCheckerGit/Console/Command/SuggestCommand.php (hunk 2 above)
Comment (mallardduck) [reply src/PHPSemVerCheckerGit/Console/Command/SuggestCommand.php]: Resolved now. This file is untouched.

**Comment by mallardduck** [comment]
A @tomzx I think I should have the gitter package PR in a good spot now.
Similar to this package it's only supporting PHP 7.2+, so helped fix this PR up.
See: https://github.com/tomzx/gitter/pull/4

**Comment by tomzx** [comment]
Hey @mallardduck,

I've done the upgrade to 0.14 slightly differently. Instead of upgrading gitter, I've simply kept it as is, but I forced `symfony/process` to `^3.4`. I've also updated the `vierbergenlars/php-semver` dependency constraint to `^3.0.2`.

This seems to resolve the issue mentioned in https://github.com/tomzx/php-semver-checker-git/issues/29.

I've pushed those changes onto master. Let me know if that solves your issue or not.

**Comment by mallardduck** [comment]
Hey @tomzx - I've just now been able to take a bit of time to test this out.
I pulled a fresh copy of `master` branch from upstream (your repo) and it mostly works.

However one major issue is how execution is handled when no suggestions.
It also seems to be specific to repos in an odd way I wouldn't expect.

### Suggest Error on Master Branch
```
Testing HEAD against tag: 0.1.0
PHP Fatal error:  Uncaught TypeError: Argument 2 passed to PHPSemVerCheckerGit\Filter\SourceFilter::filter() must be of the type array, null given, called in phar:///Users/danpock/GitProjects/php-semver-checker-git/php-semver-checker-git.phar/src/PHPSemVerCheckerGit/SourceFileProcessor.php on line 78 and defined in phar:///Users/danpock/GitProjects/php-semver-checker-git/php-semver-checker-git.phar/src/PHPSemVerCheckerGit/Filter/SourceFilter.php:7
Stack trace:
#0 phar:///Users/danpock/GitProjects/php-semver-checker-git/php-semver-checker-git.phar/src/PHPSemVerCheckerGit/SourceFileProcessor.php(78): PHPSemVerCheckerGit\Filter\SourceFilter->filter(Array, NULL)
#1 phar:///Users/danpock/GitProjects/php-semver-checker-git/php-semver-checker-git.phar/src/PHPSemVerCheckerGit/Console/Command/SuggestCommand.php(94): PHPSemVerCheckerGit\SourceFileProcessor->processFileList('HEAD', 'src', NULL)
#2 phar:///Users/danpock/GitProjects/php-semver-checker-git/php-semver-checker-git.phar/vendor/symfony/console/Command/Command.php(255): PHPSemVe in phar:///Users/danpock/GitProjects/php-semver-checker-git/php-semver-checker-git.phar/src/PHPSemVerCheckerGit/Filter/SourceFilter.php on line 7

Fatal error: Uncaught TypeError: Argument 2 passed to PHPSemVerCheckerGit\Filter\SourceFilter::filter() must be of the type array, null given, called in phar:///Users/danpock/GitProjects/php-semver-checker-git/php-semver-checker-git.phar/src/PHPSemVerCheckerGit/SourceFileProcessor.php on line 78 and defined in phar:///Users/danpock/GitProjects/php-semver-checker-git/php-semver-checker-git.phar/src/PHPSemVerCheckerGit/Filter/SourceFilter.php:7
Stack trace:
#0 phar:///Users/danpock/GitProjects/php-semver-checker-git/php-semver-checker-git.phar/src/PHPSemVerCheckerGit/SourceFileProcessor.php(78): PHPSemVerCheckerGit\Filter\SourceFilter->filter(Array, NULL)
#1 phar:///Users/danpock/GitProjects/php-semver-checker-git/php-semver-checker-git.phar/src/PHPSemVerCheckerGit/Console/Command/SuggestCommand.php(94): PHPSemVerCheckerGit\SourceFileProcessor->processFileList('HEAD', 'src', NULL)
#2 phar:///Users/danpock/GitProjects/php-semver-checker-git/php-semver-checker-git.phar/vendor/symfony/console/Command/Command.php(255): PHPSemVe in phar:///Users/danpock/GitProjects/php-semver-checker-git/php-semver-checker-git.phar/src/PHPSemVerCheckerGit/Filter/SourceFilter.php on line 7
```

Command Run:
```
± |master ✓| → ../php-semver-checker-git/php-semver-checker-git.phar suggest --allow-detached -vvv --details --include-before=src --include-after=src
```

Repo State:
Screen Shot 2020-12-15 at 6 04 25 PM


### Expected results on my PR branch
```
Testing HEAD against tag: 0.1.0

Initial semantic version: 0.1.0
Suggested semantic version: 0.1.0

Suggested semantic versioning change: NONE

[Scanned files] Before: 0 (1722 unfiltered), After: 0 (1722  unfiltered)
Time: 0.192 seconds, Memory: 10.592 MB
```

Command Run:
```
± |master ✓| → ../php-semver-checker-git/php-semver-checker-git.new.phar suggest --allow-detached -vvv --details --include-before=src --include-after=src
```

NOTE: "php-semver-checker-git.new.phar" is one I generated with box on Nov 16th.

However - as mentioned the same error doesn't happen on a different repo.
The main difference between the repos being examined is age of repo and status.
I.e. the repo with the issue is brand new and isn't pushed to remote yet at all.

### Working suggest on same repo without current tag
Also interesting context - the error only happens on this repo if the suggestion is to do nothing.
For instance if I delete the test tag I made of `0.1.0` then I see expected suggest output like:
```
Testing HEAD against tag: 0.0.1

Initial semantic version: 0.0.1
Suggested semantic version: 0.1.0

Suggested semantic versioning change: MAJOR

Class (MAJOR)
+-------+------------------------+------------------------------------------------------------+-----------------------------------+------+
| Level | Location               | Target                                                     | Reason                            | Code |
+-------+------------------------+------------------------------------------------------------+-----------------------------------+------+
| MAJOR | src/ImmutaFopen.php:35 | MallardDuck\ImmutaFopen\ImmutaFopen::recycleAtBytePosition | [public] Method has been removed. | V006 |
| PATCH | src/ImmutaFopen.php:8  | MallardDuck\ImmutaFopen\ImmutaFopen                        | Class name case was changed.      | V154 |
+-------+------------------------+------------------------------------------------------------+-----------------------------------+------+

[Scanned files] Before: 1 (1722 unfiltered), After: 1 (1722  unfiltered)
Time: 0.349 seconds, Memory: 11.312 MB
```

Command Run:
```
± |master ✓| → ../php-semver-checker-git/php-semver-checker-git.phar suggest --allow-detached -vvv --details --include-before=src --include-after=src
```

Repo State:
Screen Shot 2020-12-15 at 6 02 35 PM

**Comment by tomzx** [comment]
@mallardduck To confirm, the error you list under `Suggest Error on Master Branch` is using a newly built `php-semver-checker-git.phar` using the master branch, is that correct? Just asking because you are not using the `php-semver-checker-git.new.phar` you mentioned for that command.

**Comment by mallardduck** [comment]
Hey @tomzx to clarify:

- Dec 15 17:51 php-semver-checker-git.phar - Based on Master
- Nov 16 21:36 php-semver-checker-git.new.phar - Based on my PR

Updated the comment a bit to hopefully make it more clear.

**Comment by tomzx** [comment]
Thanks for the clarification!

It looks like this might be due to the update of a dependency in `php-semver-checker`that returns `null` instead of an empty string. I'll see if I can replicate on my side.

In the meantime if you can create a small case worth reproductible steps, that would be appreciated.

**Comment by tomzx** [comment]
@mallardduck I've pushed a fix for the issue you reported on master. Please give it a try and let me know if it works for you.

**Comment by mallardduck** [comment]
After pulling master branch and rebuilding the phar with box this looks good!
Thanks for the hard work getting this working. Gonna close this out.
I think this - hopefully - means you could tag a stable release of the *-git phar!

**Comment by tomzx** [comment]
> I think this - hopefully - means you could tag a stable release of the *-git phar!

Will do!

**Comment by tomzx** [comment]
Sorry for the delay, you can find the latest phar @ https://psvcg.coreteks.org/v0.8.0/php-semver-checker-git.phar.
